React/Views/RCTNavItem.m

```diff
@@ -11,10 +11,21 @@
 
 @implementation RCTNavItem
 
+@synthesize titleImageView = _titleImageView;
 @synthesize backButtonItem = _backButtonItem;
 @synthesize leftButtonItem = _leftButtonItem;
 @synthesize rightButtonItem = _rightButtonItem;
 
+- (UIImageView *)titleImageView
+{
+  if (_titleImage) {
+    _titleImageView = [[UIImageView alloc] initWithImage:_titleImage];
```

Comment:
Since we're always recreating it, it doesn't seem to be useful to keep it around as an instance variable. Could you remove the `@synthesize` and just return the UIImageView immediately?

I'm assuming this will be efficient enough since it only gets called on viewWillAppear. If not, you need to figure out a way to reuse the `_titleImageView` here if the image didn't change.